File: android/src/main/kotlin/com/amplitude/amplitude_flutter/FlutterLibraryPlugin.kt
package com.amplitude.amplitude_flutter

import com.amplitude.core.Amplitude
import com.amplitude.core.events.BaseEvent
import com.amplitude.core.platform.Plugin

class FlutterLibraryPlugin(val library: String): Plugin {
    override val type: Plugin.Type = Plugin.Type.Enrichment
    override lateinit var amplitude: Amplitude

    override fun execute(event: BaseEvent): BaseEvent? {
        if (event.library == null) {
            event.library = library
        } else {
            event.library = "${library}_${event.library}"
        }
        return super.execute(event)
    }
}

File: darwin/Classes/FlutterLibraryPlugin.swift
import Foundation
import AmplitudeSwift

class FlutterLibraryPlugin: BeforePlugin {
    let library: String

    init(library: String) {
        self.library = library
    }

    override func execute(event: BaseEvent) -> BaseEvent? {
        if let eventLibrary = event.library {
            event.library = "\(library)_\(eventLibrary)"
        } else {
            event.library = library
        }
        return event
    }
}
